docs/Gemfile

```diff
@@ -1,4 +1,2 @@
-source "https://rubygems.org"
-
-gem 'jekyll'
-gem 'github-pages'
+source 'https://rubygems.org'
+gem 'github-pages', group: :jekyll_plugins
```

Comment:
I've run some tests and this line is the problem. Is there any good reason to change it?
For some reason, it causes the custom Liquid extensions to not be recognised (hence the `eval` tag not recognised). But the custom filters don't work either, even though that doesn't produce an error (!!), and thus the shuffle panels are not generated... 

Reverting this line and rerunning `bundle update` updates without problem

Reply:
Ok, I got it now: I was fixing, that github would be able to build the page (so that eventually the page is also available under https://pmd.github.io/pmd/) - but it failed. I now know the reason: custom plugins (wasn't aware that eval is actually such a plugin). I've updated the Gemfiles according to the docu on github pages, but it should make no difference (e.g. the gem github-pages depends on jekyll, too, so it's technically not needed to require jekyll explicitly). 

Custom Plugins are not possible with github pages -> https://jekyllrb.com/docs/plugins/

So, I see two options:
* Avoiding custom plugins (It's probably not so good to restrict ourselves in that way)
* Changing how we publish github pages, e.g. the build scripts on travis could do this: run jekyll (they actually do this now), take the generated site, forcibly push these files to the branch "gh-pages" and we let publish github from that branch again. The source of the site would still be the "docs/" folder on the master branch.

Reply:
I wasn't aware of that. I think it's best to find a solution that would allow us to use custom plugins. Bare Liquid has extremely limited expressivity, and using it would blow up our generation logic, or make some things altogether impossible. 

Building the site before pushing it to github pages seems to be the only solution. Github pages only allows a selected few plugins to be added, so we can't even bundle a new plugin. 

But we'd have to take care, that github pages doesn't rebuild the site and overwrite everything. In that case, wouldn't the source of the site be the docs/ folder on the `gh-pages` branch rather than master?

Reply:
That should be no problem, since we would then only publish the generated html files (so, everything that was generated in `_site`).

Reply:
@adangel But wouldn't it work already with the current scripts ? Travis builds the site and pushes the already generated doc to sourceforge or to the main website (for releases), so AFAIK the build is never done by github pages.